Source/missiles.cpp

```diff
@@ -970,16 +974,16 @@ bool MonsterTrapHit(int m, int mindam, int maxdam, int dist, missile_id t, bool
 		return false;
 
 	missile_resistance mir = MissilesData[t].mResist;
-	int mor = monster.mMagicRes;
-	if (((mor & IMMUNE_MAGIC) != 0 && mir == MISR_MAGIC)
-	    || ((mor & IMMUNE_FIRE) != 0 && mir == MISR_FIRE)
-	    || ((mor & IMMUNE_LIGHTNING) != 0 && mir == MISR_LIGHTNING)) {
+	MonsterResists mor = monster.mResists;
+	if ((mor.isMagicImmune() && mir == MISR_MAGIC)
+	    || (mor.isFireImmune() && mir == MISR_FIRE)
+	    || (mor.isLightningImmune() && mir == MISR_LIGHTNING)) {

```

Comment:
Maybe we can even avoid the distinction between resistance and immunity.
```suggestion
	if (mor.getResist(mir) >= 100) {
```

Reply:
Yeah I think that makes sense, I'll modify.